src/Linters/NoPHPArrayLiteralsLinter.hack

```diff
@@ -0,0 +1,159 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\Linters;
+
+use type Facebook\HHAST\{
+  ArrayCreationExpression,
+  ArrayIntrinsicExpression,
+  ArrayToken,
+  DictToken,
+  EditableList,
+  EditableNode,
+  ElementInitializer,
+  LeftBracketToken,
+  LeftParenToken,
+  ListItem,
+  RightBracketToken,
+  RightParenToken,
+  VecToken,
+};
+use namespace HH\Lib\{C, Vec};
+use function Facebook\HHAST\Missing;
+
+final class NoPHPArrayLiteralsLinter extends AutoFixingASTLinter<EditableNode> {
```

Comment:
Can you add a common interface for ArrayCreationExpression and ArrayIntrinsicExpression (similar to src/nodes/I*php) to avoid the need to visit all EditableNodes)

Reply:
`IPHPArray` is now a thing